app/api/v1/helpers_v1.rb

```diff
@@ -12,11 +12,19 @@ def find_object(klass, id)
     error!("#{klass} not found", 404) unless @object = klass.constantize.find(id)
   end
 
-  def find_objects(klass, ids)
+  def find_objects(klass, params)
     klass = klass.classify
 
-    if ids.any?
-      @objects = klass.constantize.where(id: ids)
+    if params[:ids].any?
+      @objects = klass.constantize.where(id: params[:ids])
+    elsif params[:query] && params[:query].length > 0 && params[:limit] == 1
```

Comment:
I feel like setting a default of: 1 on the "optional :limit, type: Integer" might be preferable here. This appears to limit a custom query to a SQL LIMIT of: 1